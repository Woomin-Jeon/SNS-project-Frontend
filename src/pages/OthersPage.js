import React, { useEffect } from 'react';
import func from '../function';
import OthersPageTop from "../web_components/OthersPageTop";
import Header from "../web_components/Header";
import OthersPageSide from "../web_components/OthersPageSide";
import OthersPagePost from "../web_components/OthersPagePost";

const callAPI = async (loginState, setLoginState, currentUserState, setCurrentUserState) => {
  const { user } = await func.checkSessionExist();
  const { userStore } = await func.getUsers();

  setLoginState({
    ...loginState,
    users: [...userStore],
  });

  setCurrentUserState({
    ...currentUserState,
    id: user.id,
    userName: user.userName,
    friends: user.friends,
    profile: user.profile,
  });
};

function OthersPage({
  specificPost,
  loginState,
  setLoginState,
  currentUserState,
  setCurrentUserState,
  postState,
  setPostState,
  commentState,
  setCommentState,
  setSearchState,
}) {
  const { users } = loginState;
  const index = users.findIndex((user) => user.id === specificPost.id);

  useEffect(() => {
    callAPI(loginState, setLoginState, currentUserState, setCurrentUserState);
  }, []);

  return (
    <div>
      {!users[index]
        ? <h3>서버로부터 데이터를 받아올 수 없습니다. 다시 로그인 해주세요.</h3>
        : (
          <div className="mypage-grid">
            <OthersPageTop
              specificPost={specificPost}
              currentUserState={currentUserState}
              setCurrentUserState={setCurrentUserState}
              loginState={loginState}
              setLoginState={setLoginState}
            />
            <Header
              loginState={loginState}
              setLoginState={setLoginState}
              currentUserState={currentUserState}
              setCurrentUserState={setCurrentUserState}
              setSearchState={setSearchState}
            />
            <div className="mypage-timeline-grid">
              <OthersPageSide
                loginState={loginState}
                specificPost={specificPost}
              />
              <OthersPagePost
                postState={postState}
                setPostState={setPostState}
                currentUserState={currentUserState}
                commentState={commentState}
                setCommentState={setCommentState}
                loginState={loginState}
                setLoginState={setLoginState}
                specificPost={specificPost}
              />
            </div>
          </div>
        )}
    </div>
  );
}

export default OthersPage;
