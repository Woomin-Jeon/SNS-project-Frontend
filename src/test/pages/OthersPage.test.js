import OthersPage from "../../pages/OthersPage";
import React from "react";
import {shallow} from 'enzyme';

describe('OthersPage', () => {
  let currentUserState;
  let setCurrentUserState;
  let loginState;
  let setLoginState;
  let commentState;
  let setCommentState;
  let postState;
  let setPostState;
  let topLevelState;
  let setSearchState;

  beforeEach(() => {
    postState = { post: [] };
    loginState = { users: [] };
  });

  it('renders well', () => {
    const component = shallow(
      <OthersPage
        specificPost={topLevelState}
        loginState={loginState}
        setLoginState={setLoginState}
        currentUserState={currentUserState}
        setCurrentUserState={setCurrentUserState}
        postState={postState}
        setPostState={setPostState}
        commentState={commentState}
        setCommentState={setCommentState}
        setSearchState={setSearchState}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
