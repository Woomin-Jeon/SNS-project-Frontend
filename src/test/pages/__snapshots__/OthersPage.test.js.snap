// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OthersPage renders well 1`] = `ShallowWrapper {}`;
